tests/playwright/src/specs/kubernetes-edit-yaml.spec.ts

```diff
@@ -0,0 +1,124 @@
+
+  await welcomePage.handleWelcomePage(true);
+  await waitForPodmanMachineStartup(page);
+  if (!skipKindInstallation) {
+    const settingsBar = await navigationBar.openSettings();
+    await settingsBar.cliToolsTab.click();
+
+    await ensureKindCliInstalled(page);
+  }
+  await createKindCluster(page, CLUSTER_NAME, true, CLUSTER_CREATION_TIMEOUT);
+});
+
+test.afterAll(async ({ runner, page }) => {
+  test.setTimeout(90000);
+  try {
+    await deleteKindCluster(page, KIND_NODE, CLUSTER_NAME);
+  } finally {
+    await runner.close();
+  }
+});
+
+test.describe('Kubernetes Edit YAML Feature E2E Test', () => {
+  test.skip(
+    !!process.env.GITHUB_ACTIONS && process.env.RUNNER_OS === 'Linux',
+    'Tests suite should not run on Linux platform',
+  );
+  test('Create a Kubernetes deployment resource', async ({ navigationBar }) => {
+    test.setTimeout(80_000);
+    const podsPage = await navigationBar.openPods();
+    await playExpect(podsPage.heading).toBeVisible();
+    const playYamlPage = await podsPage.openPlayKubeYaml();
+    await playExpect(playYamlPage.heading).toBeVisible();
+    const yamlFilePath = path.resolve(__dirname, '..', '..', 'resources', 'kubernetes', `${DEPLOYMENT_NAME}.yaml`);
+    await playYamlPage.playYaml(yamlFilePath, {
+      runtime: PlayYamlRuntime.Kubernetes,
+      kubernetesContext: KUBERNETES_CONTEXT,
+      kubernetesNamespace: KUBERNETES_NAMESPACE,
+    });
+
+    const kubernetesBar = await navigationBar.openKubernetes();
+    const deploymentsPage = await kubernetesBar.openTabPage(KubernetesResources.Deployments);
+    await playExpect(deploymentsPage.heading).toBeVisible();
+    await playExpect(deploymentsPage.getResourceRowByName(DEPLOYMENT_NAME)).toBeVisible();
+    const deploymentDetails = await deploymentsPage.openResourceDetails(
+      DEPLOYMENT_NAME,
+      KubernetesResources.Deployments,
+    );
+    await playExpect(deploymentDetails.heading).toBeVisible();
+    await playExpect
+      .poll(async () => deploymentDetails.getState(), { timeout: 80_000 })
+      .toEqual(KubernetesResourceState.Running);
+  });
+  test('Change the Kubernetes deployment YAML file', async ({ navigationBar }) => {
+    test.setTimeout(80_000);
+    const podsPage = await navigationBar.openPods();
+    await playExpect.poll(async () => await podsPage.countPodReplicas(DEPLOYMENT_NAME)).toEqual(3);
+
+    const kubernetesBar = await navigationBar.openKubernetes();
+    const deploymentsPage = await kubernetesBar.openTabPage(KubernetesResources.Deployments);
+    await playExpect(deploymentsPage.heading).toBeVisible();
+    await playExpect(deploymentsPage.getResourceRowByName(DEPLOYMENT_NAME)).toBeVisible();
+    const deploymentDetails = await deploymentsPage.openResourceDetails(
+      DEPLOYMENT_NAME,
+      KubernetesResources.Deployments,
+    );
+    await playExpect(deploymentDetails.heading).toBeVisible();
+    await deploymentDetails.editKubernetsYamlFile('replicas: 3', 'replicas: 5');
```

Comment:
I am curious what will happen when you try to replace all if it (and probably will) find more that one occurrence of the string to replace?
I think there is more than one `*replicas: 3` in the file? Can you please confirm?

Reply:
Yes, there are more `*replicas: 3`. However, other than the `spec.replicas` are just metadata, so editing them doesn't affect anything.

Reply:
They will still be changed to `*replicas: 5` after applying changes only to `spec.replicas`